test/kontrol/README.md

```diff
@@ -0,0 +1,200 @@
+
+### Kontrol usage
+
+[`test/kontrol/proofs`](./proofs) contains Solidity files with Kontrol tests (`*.k.sol` and `*.t.sol`) we are using in this tutorial. To build the proofs, run:
+```shell
+kontrol build
+```
+in this (`test/kontrol`) directory. The arguments this command is executed with are specified in the [`kontrol.toml`](../../kontrol.toml) file.
+
+To run the proofs, execute:
+```shell
+kontrol prove
+```
+
+A specific proof can be run by passing the `--mt` flag with the name of the proof. For example, to run the [`prove_Allowance_ReturnsZero`](./proofs/DTokenProof.k.sol) proof for the `DToken` contract, execute:
+```shell
+kontrol prove --mt prove_Allowance_ReturnsZero
+```
+This command will ensure that `DToken` contract's `allowance` function returns zero for _any_ input addresses.
+
+### Examples
+
+In this repository, you can find the following examples:
+
+#### 1. [Counter](./proofs/Counter.k.sol)
+
+Taken from the [Kontrol documentation](https://docs.runtimeverification.com/kontrol/guides/kontrol-example/property-verification-using-kontrol), `CounterTest` is a simple (but eloquent) example of how Kontrol can catch an issue missed by a fuzzer.
+Please note that, at the moment, this example is not available in the Docker container.
+
+For illustration purposes, we've modified the code of a `Counter` contact by adding a special case in which its `setNumber` function would not update the number value and will throw an error instead.
+This will happen if two input parameters have specific values: `uint256 newNumber` is `0xC0FFEE` and `bool inLuck` is `true`. In other words, a test such as 
+```solidity
+    function testSetNumber(uint256 x, bool inLuck) public {
+        counter.setNumber(x, inLuck);
+        assert(counter.number() == x);
+    }
+```
+should fail if `x` is `0xC0FFEE` and `inLuck` is `true`, and pass otherwise.
+
+To run this test with Kontrol, execute:
+```shell
+kontrol prove --match-test CounterTest.testSetNumber
+```
+
+In the output, you should see that the test has failed with the following model (i.e., counterexample) being printed:
+```
+  Path condition:
+    { VV0_x_114b9705:Int #Equals 12648430 }
+#And { VV1_inLuck_114b9705:Int #Equals 1 }
+  Model:
+    VV1_inLuck_114b9705 = 1
+    ORIGIN_ID = 645326474426547203313410069153905908525362434350
+    CALLER_ID = 645326474426547203313410069153905908525362434350
+    TIMESTAMP_CELL = 0
+    VV0_x_114b9705 = 12648430
+    NUMBER_CELL = 0
+```
+The path condition tells us that the failure occurs when `x == 12648430` and `inLuck == 1`, where `12648430` is the decimal representation of `0xC0FFEE`, and `1` represents `true` for Boolean values.
+
+The model represents concrete assignments to symbolic variables (`x`, `inLuck`, globally accessible variables such as `msg.sender` (`CALLER_ID`) and `tx.origin` (`ORIGIN_ID`)) that lead to the failure.
```

Comment:
I'd maybe rephrase this along the following lines, but feel free to leave it as is!
> The model represents concrete assignments to the symbolic variables `x` and `inLuck` that lead to the failure. It also concretizes globally accessible variables such as `msg.sender` (`CALLER_ID`) and `tx.origin` (`ORIGIN_ID`), although their values are not relevant for this example.

Reply:
Sounds great, thanks for the suggestion!